### PR TITLE
Fix provider.yaml errors due to exit(0) in test

### PR DESCRIPTION
**Pull request by potiuk**
There were a few errors in provider structure resulting from
accidental exit(0) in provider's tests. Luckily none of the changes
have been released yet.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/s3_upload.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from airflow.providers.amazon.aws.hooks.s3 import S3Hook
 
 
-class LocalFilesystemToS3Operator(BaseOperator):
+class S3UploadOperator(BaseOperator):
```
Comment (kaxil) [inline airflow/providers/amazon/aws/operators/s3_upload.py]: hmm.. -- why this change?

Comment (potiuk) [reply airflow/providers/amazon/aws/operators/s3_upload.py]: Because if we have "transfer" we need to have two integration "source" and "target" in provider.yaml (and in the documentation) . The "transfer" operators by definition take two Hooks and perform active transfer operation between them. In this case we have no "Source" integration because we are simply uploading a local file to S3.

Comment (potiuk) [reply airflow/providers/amazon/aws/operators/s3_upload.py]: See here for example: https://github.com/apache/airflow/blob/e18b6a6d19f9ea0d8fe760ba00adf38810f0e510/airflow/providers/amazon/provider.yaml#L408

Comment (potiuk) [reply airflow/providers/amazon/aws/operators/s3_upload.py]: It was simply wrongly classified type of operator. This operator is a "classic" operator not "transfer" one.

Comment (potiuk) [reply airflow/providers/amazon/aws/operators/s3_upload.py]: UPDATE: My bad. `Local` integration is way better here.

**Review by eladkal — APPROVED** [review]
